modelzoo/DBMTL/train.py

```diff
@@ -0,0 +1,921 @@
+
+    train_init_op = iterator.make_initializer(train_dataset)
+    test_init_op = iterator.make_initializer(test_dataset)
+
+    # create future column
+    feature_cols = build_feature_cols()
+
+    # create variable partitioner for distributed training
+    num_ps_replicas = len(tf_config['ps_hosts']) if tf_config else 0
+    input_layer_partitioner = partitioned_variables.min_max_variable_partitioner(
+        max_partitions=num_ps_replicas,
+        min_slice_size=args.input_layer_partitioner <<
+        20) if args.input_layer_partitioner else None
+    dense_layer_partitioner = partitioned_variables.min_max_variable_partitioner(
+        max_partitions=num_ps_replicas,
+        min_slice_size=args.dense_layer_partitioner <<
+        10) if args.dense_layer_partitioner else None
+    
+    # Session config
+    sess_config = tf.ConfigProto()
+    sess_config.inter_op_parallelism_threads = args.inter
+    sess_config.intra_op_parallelism_threads = args.intra
+
+    # Session hooks
+    hooks = []
+
+    if args.smartstaged and not args.tf:
+        '''Smart staged Feature'''
+        next_element = tf.staged(next_element, num_threads=4, capacity=40)
+        sess_config.graph_options.optimizer_options.do_smart_stage = True
+        hooks.append(tf.make_prefetch_hook())
+    if args.op_fusion and not args.tf:
+        '''Auto Graph Fusion'''
+        sess_config.graph_options.optimizer_options.do_op_fusion = True
+    if args.micro_batch and not args.tf:
+        '''Auto Mirco Batch'''
+        sess_config.graph_options.optimizer_options.micro_batch_num = args.micro_batch
+    
+    # create model
+    model = DBMTL(input=next_element,
+                  feature_column=feature_cols,
+                  bottom_dnn=BOTTOM_DNN,
+                  towers=TOWERS,
+                  relation_dnn=RELATION_DNN,
+                  optimizer_type=args.optimizer,
+                  bf16=args.bf16,
+                  stock_tf=args.tf,
+                  adaptive_emb=args.adaptive_emb,
+                  input_layer_partitioner=input_layer_partitioner,
+                  dense_layer_partitioner=dense_layer_partitioner)
+    
+    # run model training and evaluation
+    train(sess_config, hooks, model, train_init_op, train_steps,
+          checkpoint_dir, tf_config, server)
+    if not (args.no_eval or tf_config):
+        eval(sess_config, hooks, model, test_init_op, test_steps,
+             checkpoint_dir)
+
+def boolean_string(string):
+    low_string = string.lower()
+    if low_string not in {'false', 'true'}:
+        raise ValueError('Not a valid boolean string')
+    return low_string == 'true'
+
+def get_arg_parser():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    # parser.add_argument('--job',
+    #                     help='train or test model',
+    #                     type=str,
+    #                     choices=["train", "test"],
+    #                     default='train')
```

Comment:
Clean up comments

Reply:
Cleaned up unnecessary comments, there are still some that I need for WIP.